libs/feature/editor/src/lib/services/editor.service.ts

```diff
@@ -20,12 +20,16 @@ export class EditorService {
   ): Observable<[CatalogRecord, string]> {
     const savedRecord = { ...record }
 
+    const fields = fieldsConfig.pages
+      .map((page) => page.sections.map((section) => section.fields))
+      .flat(2)
```

Comment:
2 because? (I love the no-magical-number eslint rule :D)

Always think of the developers coming afterwards (might be you in 3 months). Anything that helps understand the code with less context is beneficial.